modules/vye/spec/models/vye/user_info_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Vye::UserInfo, type: :model do
+  before do
+    stub_const 'Vye::GenDigest::Common::GEN_DIGEST_CONFIG', Vye::GenDigest::Common.get_gen_digest_config
+  end
+
+  describe 'create' do
+    it 'creates a new record' do
+      expect do
+        attributes = FactoryBot.attributes_for(:vye_user_info)
+        Vye::UserInfo.create!(attributes)
+      end.to change(Vye::UserInfo, :count).by(1)
+    end
+  end
+
+  describe 'find_by_user after ICN is recorded' do
+    let!(:user) { create(:evss_user, :loa3) }
+    let!(:user_info) { FactoryBot.create(:vye_user_info, icn: user.icn) }
+
+    it 'finds the user info by icn' do
+      u = Vye::UserInfo.find_and_update_icn(user:)
+      expect(u).to eq(user_info)
+    end
+  end
+
+  describe 'find_by_user before ICN is recorded' do
+    let!(:user) { create(:evss_user, :loa3) }
+    let!(:user_info) { FactoryBot.create(:vye_user_info, icn: nil, ssn: user.ssn) }
+
+    it 'finds the user info by ssn' do
+      u = Vye::UserInfo.find_and_update_icn(user:)
+      expect(u).to eq(user_info)
+      expect(u.icn_in_database).to eq(user.icn)
```

Comment:
You'll want to duplicate these tests in order to test the code with the flipper on or the flipper off. Before the test, define whether the flipper is on or off. Search specs for `Flipper.enable` for examples.

Reply:
done

Reply:
@binq I see you've enabled the flipper before the tests but I don't see any tests for when the flipper is disabled.